src/backend/metal/src/device.rs

```diff
@@ -294,6 +294,11 @@ impl Device {
                 arg.set_data_type(MTLDataType::Texture);
                 arg.set_index(index as NSUInteger);
             }
+            DescriptorType::UniformBuffer | DescriptorType::StorageBuffer => {
+                arg.set_access(MTLArgumentAccess::ReadOnly);
```

Comment:
I don't think storage buffers are read-only.

Reply:
@msiglreith  Can you explain a bit the difference between `UniformBuffer` and `StorageBuffer`?

Reply:
`UniformBuffer`: Read-only structured buffer
`StorageBuffer`: Read-Write structured buffer (+atomics)